### ISSUE TITLE
Full VM testnet: Terraform does not return public and yggdrasil IP

### ISSUE THREAD

**Issue by coesensbert** [threefoldtech/terraform-provider-grid#205]
### Terraform Version
```
➜  test git:(master) ✗ terraform -v            
Terraform v1.1.7
on linux_amd64
+ provider registry.terraform.io/threefoldtech/grid v0.1.26
```

### Affected Resource(s)

```
➜  test git:(master) ✗ terraform show
# grid_deployment.d1:
resource "grid_deployment" "d1" {
    id           = "3849"
    ip_range     = "10.1.2.0/24"
    network_name = "tfopsnetwork"
    node         = 7

    disks {
        name = "data"
        size = 50
    }

    vms {
        cpu         = 4
        description = "A real VM on the grid"
        env_vars    = {
            "SSH_KEY" = "ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIDYNeJXJV2FNEwuQz6e0jkKeqRbKWwftBKq+sjSTqa2x"
        }
        flist       = "https://hub.grid.tf/azmy.3bot/ubuntu-image.flist"
        ip          = "10.1.2.2"
        memory      = 8192
        name        = "realvmtest1"
        planetary   = true
        publicip    = true
        publicip6   = true
        rootfs_size = 0

        mounts {
            disk_name   = "data"
            mount_point = "/data"
        }
    }
}

# grid_network.net1:
resource "grid_network" "net1" {
    add_wg_access      = false
    description        = "threefold ops test network"
    external_sk        = "sLgHirQBGrhexepLUu1D6Cxwm71l8Ki9ChiOW2ZoRUI="
    id                 = "640f0333-8ba2-4474-972f-eaee8b47ed2f"
    ip_range           = "10.1.0.0/16"
    name               = "tfopsnetwork"
    node_deployment_id = {
        "7" = 3847
    }
    nodes              = [
        7,
    ]
    nodes_ip_range     = {
        "7" = "10.1.2.0/24"
    }
    public_node_id     = 0
}


Outputs:

node1_vm1_ip = "10.1.2.2"
public_ip = ""
public_ip6 = ""
wg_config = ""
ygg_ip = ""
```

### Terraform Configuration Files
```hcl
➜  test git:(master) ✗ cat main.tf                                                    
terraform {
  required_providers {
    grid = {
      source = "threefoldtech/grid"
    }
  }
}

provider "grid" {
}

resource "grid_network" "net1" {
    nodes = [7]
    ip_range = "10.1.0.0/16"
    name = "tfopsnetwork"
    description = "threefold ops test network"
    add_wg_access = false
}
resource "grid_deployment" "d1" {
  node = 7
  network_name = grid_network.net1.name
  ip_range = lookup(grid_network.net1.nodes_ip_range, 2, "10.1.2.0/24")
  disks {
    name = "data"
    size = 50
  }
    vms {
    name = "realvmtest1"
    description ="A real VM on the grid"
    flist = "https://hub.grid.tf/azmy.3bot/ubuntu-image.flist"
    cpu = 4
    publicip = true
    publicip6 = true
    memory = 8192
#    rootfs_size = 25000
    mounts {
        disk_name = "data"
        mount_point = "/data"
    }
    planetary = true
    env_vars = {
      SSH_KEY ="ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIDYNeJXJV2FNEwuQz6e0jkKeqRbKWwftBKq+sjSTqa2x"
    }
  }
}
output "wg_config" {
    value = grid_network.net1.access_wg_config
}
output "node1_vm1_ip" {
    value = grid_deployment.d1.vms[0].ip
}
output "public_ip" {
    value = grid_deployment.d1.vms[0].computedip
}
output "public_ip6" {
    value = grid_deployment.d1.vms[0].computedip6
}
output "ygg_ip" {
    value = grid_deployment.d1.vms[0].ygg_ip
}
```

### Expected Behavior
After deployment, return pub and ygg IP

### Actual Behavior
Deployed fine, only no pub and ygg IP. Had to check the chain with contract ID to find out the IP's.

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

**Comment by OmarElawady** [comment]
The flist link returns 404, was it removed?
https://hub.grid.tf/azmy.3bot/ubuntu-image.flist

**Comment by coesensbert** [comment]
Not sure about that specific one, but you can find the others here:

https://hub.grid.tf/tf-official-vms

Lately i'm deploying using this image: https://hub.grid.tf/tf-official-vms/ubuntu-20.04-lts.flist

**Comment by OmarElawady** [comment]
After checking with @coesensbert, it seems the issue doesn't happen anymore.

It's possible that the machine restarted multiple time and its state got converted from `created` to `deleted` (between the check that the deployment is ready and fetching its information).

In case it happens again and the machine is running properly. Doing `terraform refresh` might get the info (it would still be an issue but if the info appears it's more info). If both doesn't work, the specific deployment can be investigated then.

**Comment by OmarElawady** [comment]
I think I found the cause of this. After the deployment succeeds, the state is updated on a best-effort basis. The logic is that if the state syncing failed, it could be resynced by a refresh or a second apply so no need to destroy everything for it. But it was failing silently without producing a warning. This is fixed in v1.2.0.

The reason for failure to sync could be anything but my guess is that it's because of [this](https://github.com/threefoldtech/terraform-provider-grid/issues/277). I'll close this issue and track new errors causing this which should appear as a warning in v1.2.0 versoin.
